fiftyone/plugins/secrets.py

```diff
@@ -155,5 +168,7 @@ def values(self):
         raise RuntimeError("Iteration through values is not allowed")
 
     def items(self):
-        # Override items() to prevent iterating through plaintext values
-        raise RuntimeError("Iteration through items is not allowed.")
+        # Override items() to use __getitem__ to automatically resolve
+        # missing secrets upon iteration
+        for key in self.keys():
+            yield key, self[key]
```

Comment:
IDK how the resolver is implemented in teams, but need to verify: is calling `self[key]` in a loop performant? IE will it only make one API call to pull in all secrets?

There are plugins that declare 10+ secrets, so 10 API requests would be 😢

Reply:
this is the best option right now. Anyways, the secrets are only potentially unresolved in resolve_input(). Any usage of the secrets in execute or resolve_ouput() should already have batch resolved the secrets. Moreover, self[key] will not only resolve it, but add it to the secrets dict so the maximum number of requests is the number of unique secrets. If performance is an issue, then it is likely the number of times the context is regenerated/operators are initialized.